usr/lib/common/encr_mgr.c

```diff
@@ -1047,3 +1048,145 @@ encr_mgr_encrypt_final(STDLL_TokData_t *tokdata,
 
     return CKR_FUNCTION_FAILED;
 }
+
+CK_RV encr_mgr_reencrypt_single(STDLL_TokData_t *tokdata, SESSION *sess,
+                                ENCR_DECR_CONTEXT *decr_ctx,
+                                CK_MECHANISM *decr_mech,
+                                CK_OBJECT_HANDLE decr_key,
+                                ENCR_DECR_CONTEXT *encr_ctx,
+                                CK_MECHANISM *encr_mech,
+                                CK_OBJECT_HANDLE encr_key,
+                                CK_BYTE *in_data, CK_ULONG in_data_len,
+                                CK_BYTE *out_data, CK_ULONG *out_data_len)
+{
+    OBJECT *decr_key_obj = NULL;
+    OBJECT *encr_key_obj = NULL;
+    CK_ATTRIBUTE *attr = NULL;
+    CK_ULONG decr_data_len = 0;
+    CK_BYTE *decr_data = NULL;
+    CK_BBOOL flag;
+    CK_RV rc;
+
+    if (!sess || !decr_ctx || !encr_ctx || !decr_mech || !encr_mech) {
+        TRACE_ERROR("Invalid function arguments.\n");
+        return CKR_FUNCTION_FAILED;
+    }
+
+    if (decr_ctx->active != FALSE || encr_ctx->active != FALSE) {
+        TRACE_ERROR("%s\n", ock_err(ERR_OPERATION_ACTIVE));
+        return CKR_OPERATION_ACTIVE;
+    }
+
+    if (token_specific.t_reencrypt_single != NULL) {
```

Comment:
Yea well, in general I agree. However all the functions in encr_mgr.c are named like encr_mgr_xxx, so these 2 subroutines should then probably also be named encr_mgr_reencrypt_single_token and encr_mgr_reencrypt_single_manual, and I would also need to pass almost all parameters to these functions. So I don't really think this will make the code better readable.... 
I would prefer to leave it as is, unless others object.